packages/cli/src/databases/migrations/sqlite/1693554410387-DisallowOrphanExecutions.ts

```diff
@@ -0,0 +1,77 @@
+import type { MigrationContext, ReversibleMigration } from '@db/types';
+
+export class DisallowOrphanExecutions1693554410387 implements ReversibleMigration {
+	transaction = false as const;
+
+	/**
+	 * Ensure all executions point to a workflow. Recreate table because sqlite
+	 * does not support modifying column.
+	 */
+	async up({ escape, schemaBuilder: { createTable, column }, runQuery }: MigrationContext) {
+		const executionEntity = escape.tableName('execution_entity');
+		const workflowId = escape.tableName('workflowId');
+
+		await runQuery(`DELETE FROM ${executionEntity} WHERE ${workflowId} IS NULL;`);
+
+		await createTable('temp')
+			.withColumns(
+				column('id').int.primary.autoGenerate.notNull,
+				column('workflowId').varchar(36).notNull, // only change
+				column('finished').bool.notNull,
+				column('mode').varchar().notNull,
+				column('retryOf').varchar(),
+				column('retrySuccessId').varchar(),
+				column('startedAt').timestamp().notNull,
+				column('stoppedAt').timestamp(),
+				column('waitTill').timestamp(),
+				column('status').varchar(),
+			)
+			.withForeignKey('workflowId', {
+				tableName: 'workflow_entity',
+				columnName: 'id',
+				onDelete: 'CASCADE',
+			});
+
+		await runQuery(
+			`INSERT INTO "temp" ("id", "workflowId", "finished", "mode", "retryOf", "retrySuccessId", "startedAt", "stoppedAt", "waitTill", "status") SELECT "id", "workflowId", "finished", "mode", "retryOf", "retrySuccessId", "startedAt", "stoppedAt", "waitTill", "status" FROM ${executionEntity};`,
+		);
+
+		await runQuery(`DROP TABLE ${executionEntity};`);
```

Comment:
`await dropTable('execution_entity').execute(queryRunner);` fails with `TypeError: Cannot read properties of undefined (reading 'name')`.

Reply:
the `.execute(queryRunner)` bit seems to be from an early version of the DSL, and shouldn't be in the code.

Reply:
I was seeing something like this yesterday, and after debugging for an hour I realized that this was a bug in typeorm, and upgrading fixed the issue.
That's why I have upgraded typeorm in https://github.com/n8n-io/n8n/pull/7073